src/http/common.cr

```diff
@@ -71,11 +71,13 @@ module HTTP
     end
   end
 
+  DEFAULT_MIME_TYPE = MIME::MediaType.parse("application/octet-stream")
+
   private def self.check_content_type_charset(body, headers)
     return unless body
 
     if content_type = headers["Content-Type"]?
-      mime_type = MIME::MediaType.parse(content_type)
+      mime_type = MIME::MediaType.parse?(content_type) || DEFAULT_MIME_TYPE
```

Comment:
`mime_type` is not used anywhere except for querying the `charset` parameter (which isn't set in `application/octet-stream` anyway). So the following condition should simply be skipped when `parse`?  returns `nil`.

Reply:
ok, so something like that?
```cr
    if (content_type = headers["Content-Type"]?) &&
       (mime_type = MIME::MediaType.parse?(content_type)) &&
       (charset = mime_type["charset"]?)
      body.set_encoding(charset, invalid: :skip)
    end
```

Reply:
Why not just `return unless mime_type`?

Reply:
Yes. In fact we should `return unless` a lot more here, to avoid nested code.